airflow/utils/log/log_reader.py

```diff
@@ -0,0 +1,104 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import logging
+from typing import Any, Dict, List, Optional, Tuple
+
+from cached_property import cached_property
+
+from airflow.configuration import conf
+from airflow.models import TaskInstance
+from airflow.utils.helpers import render_log_filename
+
+
+class TaskLogReader:
+    """ Task log reader"""
+
+    def read_log_chunks(self, ti: TaskInstance, try_number: Optional[int],
+                        metadata) -> Tuple[List[str], Dict[str, Any]]:
+        """
+         Reads logs in chunks
+        :param ti: The taskInstance
+        :type ti: TaskInstance
+        :param try_number: The taskInstance try_number
+        :type try_number: Optional[int]
+        :param metadata: A dictionary containing information about how to read the task
+        :type metadata: dict
+
+        """
+
+        logs, metadatas = self.log_handler.read(ti, try_number, metadata=metadata)
+        metadata = metadatas[0]
+        return logs, metadata
+
+    def read_log_stream(self, ti: TaskInstance, try_number: Optional[int], metadata: dict):
```

Comment:
Can you add return type?

Reply:
Fixed https://github.com/apache/airflow/pull/9391/commits/f8298101d216b62a3c598f444e5939f220c813b5